### PR TITLE
feat: Generate PTID in Agent

### PR DESCRIPTION
**Pull request by metal-messiah**
Generate PTID in agent instead of in RUM response.  This allows PageView events to be decorated with a page trace ID.
---

### Overview
This abstracts the action of generating our own PTID out of the ST merge PR into a standalone PR

### Related Issue(s)
NR-166514
NR-167098

### Testing
Affected tests have been updated

### CODE REVIEW

**Review by patrickhousley — APPROVED** [review]
Approved but I wonder if there is more we can do around not needing to wait for the response on the first STN call in the agent. Maybe some code we could remove as well. Not necessary in this PR.

**Review by patrickhousley — APPROVED** [review]

File: src/features/session_trace/aggregate/index.js
Comment (cwli24) [inline src/features/session_trace/aggregate/index.js:198]: Use `everSent` to keep sync with L220
```suggestion
    if (result.sent && !result.failed && !this.everSent) { // continue interval harvest only after first call
```

File: src/features/session_trace/aggregate/index.js
Comment (cwli24) [inline src/features/session_trace/aggregate/index.js:103]: Why having to pass empty object in? `runHarvest` needs fixing its opts param if undefined

File: src/features/session_trace/aggregate/index.js
Comment (metal-messiah) [inline src/features/session_trace/aggregate/index.js:198]:
```suggest
if (result.sent && !result.failed && !this.#scheduler.started) { // continue interval harvest only after first call
```

Comment (metal-messiah) [reply src/features/session_trace/aggregate/index.js:198]: I need to undo this suggestion because it breaks.  everSent is set to true before sending, making this statement always falsy if we use it here.

File: src/features/session_trace/aggregate/index.js
Comment (metal-messiah) [reply src/features/session_trace/aggregate/index.js:103]: functionally identical, doesn't matter to me.

**Review by cwli24 — APPROVED** [review]
Gogogo before Ashley slams the gate

**Pull request merged**
